We want to know orthologous, paralogous and xneologous relationships of genes in a group of species.
- choose a good orthogroup, each gene in the orthogroup can represent a species
- align all sequences in this orthogroup
- draw gene tree according to the multiple sequence alignment
- revise the tree
- infer orthologous/paralogous/xenologous relationships between genes according to the tree

A 'taxon' (pl. 'taxa') is a taxonomic group at any rank.

homologs: sequences have common ancestry.
Orthologs separated by a speciation event.
Paralogs separated by a duplication event.
Xenologs separated by a horitontal gene transfer (HGT) event.
Orthology and paralogy are the most frequent homologs.

species phylogeny
gene phylogeny

Detecting orthologous, paralogous and xenologs relationships requires a species tree and a gene tree for the gene family in question.

Orthogroup: the set of genes that are descended from a single gene in the last common ancestor of all the species being considered.
Predicted orthogroups can be thought of as predicted gene/protein families.

> Written with [StackEdit](https://stackedit.io/).
<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE1NjA0MDIzNDYsOTg2MzU0MjExLC00Mj
Y0MTY0ODgsNjM2NDA1MDY0LDEwODgzODIxOTQsLTIwNjE0OTM5
MjMsLTQzMTMwODA2MCwyMDQ0OTg1Njg3LDczMDk5ODExNl19
-->