
# Predicting orthogroups
## Objective
ues BLAST and OrthoFinder to delimit orthogroups across genomes
## Background
orthogroup: the set of genes descended from a single gene in the last common ancestor of all the species being considered. An orthogroup includes orthologs paralogs and xenologs.
orthologs: sequences diverged due to a speciation event.
paralogous: sequences diverged due to duplication within a single genome.
## Pipeline
OrthoFinder=BLAST+MCL
use proteins for comparisons by BLAST (more sensitive-20 vs 4, more accurate-possibility for BLAST to infer biologically implausible frameshifting indels, as BLASTN might occasionally)
Assess quality
Parameters:  
inflation parameter
BLAST parameters
## Problem
`orthofinder -f default -t 4 -og`
-f default: working directory for this analysis. Input (Fasta files) should be here.
-t 4: during the analysis, use up to 4 threads.
-og: stop after inferring orthogroups. Do not proceed to reconstruct gene or species trees.
_QUESTION 1.  Are your orthogroups the same as those of McGill and Barker (2017)? If they differ, can you guess why this might be?_

inflation parameter: 1.5 or 1.4
_QUESTION 2.  Do your OrthoFinder results with inflation value 1.4 differ from those with inflation value 1.5?_

Select a purposeful sample of several orthogroups to Select a purposeful sample of several orthogroups
* a large orthogroup
* a small orthogroup
* an orthogroup found only in one species
* an orthogroup found in all species
* an orthogroup with different copy numbers in different species

a large orthogroup, a small orthogroup, an orthogroup found only in one species, an orthogroup found in all species, an orthogroup with different copy numbers in different species
Look for internal consistency in the orthogroup, using three approaches:

* BLAST. If you compare one member of a group with another, is there a high BLAST score? Do the alignments strongly suggest homology?
* UniProt descriptions. Search for protein accession numbers at UniProt online (http://www.uniprot.org). Do these descriptions suggest functional similarity among the orthogroup members?
* Pfam domains. Search for matches to protein sequences at Pfam online (http://pfam.xfam.org). Do these suggest functional similarity among the orthogroup members?

As well as some level of internal consistency within an orthogroup, we expect differences between orthogroups. 
For at least two orthogroups, first identify a similar sequence in a different orthogroup, using BLAST at the command-line. Does this BLAST alignment, across orthogroups, have a worse score and worse-looking alignment than those you examined within the orthogroup? Do the different orthogroups have noticeably different Uniprot descriptions and Pfam domains?
_QUESTION 3.  On the basis of your investigations, do you consider orthogroups (at your chosen inflation value) to be of high quality? Give reasons._

Try some other values for the inflation parameter. Evaluate as above.
_QUESTION 4. For the three genome-wide protein sets you used as input, if you were to perform a “downstream” analysis depending on accurate delimitation of orthogroups, which value would you use for the inflation parameter? Why?_

_QUESTION 4. What is the BLAST+ default protein substitution matrix_
`orthofinder -f 1.6 -I 1.6 -t 4 -op`

`blastp -outfmt 6 -evalue 0.001 -query /localdisk/home/dbarker/CEG/week3/1.5/Results_Jan28/WorkingDirectory/Species0.fa -db /localdisk/home/dbarker/CEG/week3/1.5/Results_Jan28/WorkingDirectory/BlastDBSpecies0 -out /localdisk/home/dbarker/CEG/week3/1.5/Results_Jan28/WorkingDirectory/Blast0_0.txt`

_QUESTION 5. For the three genome-wide protein sets you used as input, if you were to perform a “downstream” analysis depending on accurate delimitation of orthogroups, which protein substitution matrix would you use? Why?_

_QUESTION 6. Which of these two sets of orthogroups is better? Use your notes on your evaluations to give detailed reasons for your answer._



> Written with [StackEdit](https://stackedit.io/).
<!--stackedit_data:
eyJoaXN0b3J5IjpbMzIwMzQxMTAyXX0=
-->